### ISSUE TITLE
linter: `react/jsx-boolean-value` rule doesn't support "always" | "never"

### ISSUE THREAD

**Issue by arsinclair** [oxc-project/oxc#21061]
### What version of Oxlint are you using?

1.58.0

### What command did you run?

`oxlint --type-aware`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

<details><summary>oxlint config</summary>

```ts
import { defineConfig } from "oxlint";

export default defineConfig({
    plugins: ["eslint", "typescript", "unicorn", "oxc", "react", "nextjs", "import", "jsx-a11y"],
    categories: {
        correctness: "warn",
        suspicious: "warn",
        style: "warn",
        perf: "warn",
        pedantic: "off",
        restriction: "off"
    },
    env: {
        browser: true,
        es6: true,
        node: true
    },
    rules: {
        "eslint/no-restricted-imports": [
            "error",
            {
                paths: ["next/link"]
            }
        ],
        "eslint/no-unused-vars": "warn",
        "react/jsx-key": "error",
        "react/jsx-no-duplicate-props": "error",
        "react/no-unescaped-entities": "warn",
        "react/react-in-jsx-scope": "off",
        "import/no-named-export": "off",
        "eslint/sort-imports": "off",
        "eslint/sort-keys": "off",
        "import/prefer-default-export": "off",
        "eslint/func-style": "off",
        "eslint/max-statements": "off",
        "eslint/no-magic-numbers": "off",
        "eslint/no-ternary": "off",
        "import/group-exports": "off",
        "unicorn/filename-case": "off",
        "unicorn/no-null": "off",
        "eslint/capitalized-comments": "off",
        "eslint/id-length": "off",
        "import/exports-last": "off",
        "import/no-unassigned-import": "off",
        "eslint/new-cap": "off",
        "react/jsx-max-depth": "off",
        "react/jsx-props-no-spreading": "off",
        "unicorn/prefer-global-this": "off",
        "unicorn/prefer-ternary": "off",
        "import/consistent-type-specifier-style": "off",
        "eslint/curly": "off",
        "eslint/arrow-body-style": "off",
        "typescript/array-type": "off",
        "typescript/no-inferrable-types": "off",
        "react/jsx-boolean-value": "always",
        "eslint/init-declarations": "off",
        "eslint/no-shadow": "off",
        "react-hooks/exhaustive-deps": "off",
        "unicorn/consistent-function-scoping": "off",
        "eslint/no-await-in-loop": "off",
        "eslint/no-continue": "off",
        "eslint/no-nested-ternary": "off",
        "unicorn/prefer-spread": "off",
        "unicorn/prefer-array-index-of": "off",
        "eslint/prefer-destructuring": "off",
        "typescript/consistent-type-definitions": "off",
        "eslint/prefer-template": "off",
        "eslint/operator-assignment": "off",
        "eslint/max-params": "off",
        "import/no-nodejs-modules": "off",
        "unicorn/no-new-array": "off",
        "typescript/no-extraneous-class": "off"
    },
    ignorePatterns: [".next/**", "node_modules/**", ".git/**"]
});
```

</details> 

### What happened?

According to docs, the `react/jsx-boolean-value` can be set to `always` to ensure boolean props receive a boolean attribute explicitly, e.g.

`const Hello = <Sidebar isOpen={true} />;`

https://oxc.rs/docs/guide/usage/linter/rules/react/jsx-boolean-value.html

However, setting this rule to `always` produces TS warnings:

```
Type '"always"' is not assignable to type 'DummyRule'.
```

and breaks the oxlint execution:

```
> oxlint --type-aware

Failed to parse oxlint configuration file.

  × Failed to parse config from /home/user/GIT/project/oxlint.config.ts
  note: Failed to parse rule severity, expected one of "allow", "off", "deny", "error" or "warn", but got "always"
```

**Comment by camc314** [comment]
Hi, this is working as expected.


This is incorrect:
```
"react/jsx-boolean-value": "always",
```

It should be configured as:
```
"react/jsx-boolean-value": ["error", "always"],
```


Thanks

**Comment by camc314** [comment]
cc @connorshea is there an oppertunity to improve the docs here? i guess the first/second option info assums the user knows that it's an array

**Comment by arsinclair** [comment]
Thanks for your help!

Yes, it could be made more clear in the docs that `"always" | "never"` is actually the second element of an array.

**Comment by connorshea** [comment]
> cc [@connorshea](https://github.com/connorshea) is there an oppertunity to improve the docs here? i guess the first/second option info assums the user knows that it's an array

probably yeah, we could have it automatically generate an example of the config for cases like this, I guess?:

````md
## Configuration

Example config:

```json
{
  "rules": {
    "rule-name": ["error", "<first-option>", "<second-option>"]
  }
}
```

### The 1st option
````
